python/src/aiconfig/default_parsers/hf.py

```diff
@@ -115,7 +117,7 @@ def construct_regular_output(response: TextGenerationResponse, response_includes
     output = ExecuteResult(
         **{
             "output_type": "execute_result",
-            "data": response.generated_text,
+            "data": response.generated_text or '',
```

Comment:
Looking at the original implementation of this at https://github.com/lastmile-ai/aiconfig/blob/75f00d16ff64c53d0f0fdc4734eb13692bc1b1b7/python/src/aiconfig/default_parsers/hf.py#L112-L115, I notice that in certain cases (specifically if `response_includes_details` is `false`), the entire response was assigned to "data" here.

Perhaps we should check if `isinstance(response, str)` is true, in which case we should use that, otherwise use `response.generated_text`.

Reply:
This is a bit confusing, same thing as https://github.com/lastmile-ai/aiconfig/pull/611#discussion_r1437763720

I will update the comments in code here to make it more clear though